openfermioncirq/gates/four_qubit_gates.py

```diff
@@ -12,13 +12,59 @@
 
 """Gates that target four qubits."""
 
-from typing import Optional, Union
+
+from typing import Optional, Union, Tuple
 
 import numpy
 
 import cirq
 
 
+def state_swap_eigen_component(x: str, y: str, sign: int = 1):
+    """The +/- eigen-component of the operation that swaps states x and y.
+
+    For example, state_swap_eigen_component('01', '10', sign) returns
+        ┌                             ┐
+        │0, 0,           0,          0│
+        │0, 0.5,         0.5 * sign, 0│
+        │0, 0.5 * sign,  0.5,        0│
+        │0, 0,           0,          0│
+        └                             ┘
```

Comment:
For consistency, just format the matrix the same way as in the common gate docs, i.e. like
```
[1 0          0           0]
[0 cos(π·h/2) -sin(π·h/2) 0]
[0 sin(π·h/2) cos(π·h/2)  0]
[0 0          0           1]
```

Reply:
This method returns the eigen-components, not the gate itself.

Reply:
I actually prefer my way; I think the corners in your way extend a little too far upwards and downwards, and the commas are ugly. I also think you can improve it by using `±1` instead of `sign`.

Reply:
It's fine. Just end the preceding paragraph with two colons `::` and a line space after it so that the html docs are rendered properly.